### PR TITLE
Update electron regression/calibration

### PR DESCRIPTION
**Pull request by dabelknap**

### CODE REVIEW

**Comment by dabelknap** [comment]
@ekfriis this is the error I am getting with the changes I have made. I believe it is because the eleRegressionEnergy producer takes cleanPatElectrons, and I'm guessing that the cleanPatElectron collection does not yet exist when this is called. However, I can't seem to find where process.cleanPatElectrons are added to the sequence.

```
----- Begin Fatal Exception 17-Jun-2013 03:41:01 CDT-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1 lumi: 3000 event: 899865
   [1] Running path 'zzSkim'
   [2] Calling event method for module RegressionEnergyPatElectronProducer/'eleRegressionEnergy'
Exception Message:
Principal::getByLabel: Found zero products matching all criteria
Looking for type: std::vector<pat::Electron>
Looking for module label: cleanPatElectrons
Looking for productInstanceName:

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

File: PatTools/python/patElectronProduction_cff.py [hunk 1]
```diff
@@ -64,5 +67,6 @@
 customizeElectronSequence += patElectronMVAIDEmbedder
 customizeElectronSequence += patElectronMVAIDWPEmbedding
 customizeElectronSequence += patElectronsEmbedIp
-customizeElectronSequence += patElectronEnergyCorrections
+customizeElectronSequence += eleRegressionEnergy
+customizeElectronSequence += calibratedPatElectrons
```
Comment (ekfriis) [inline PatTools/python/patElectronProduction_cff.py:71]: Hi Austin,

The issue with the crash is that you need in `patTupleProduction.py` the `Utilities.chain_sequence` [1] function is called to set in the input tags to the previous sequence items output tags.  It assumes that the input should be named the standard "src", not "inputPatElectronsTag" (rage).

I will modify the cfg_chainer and send you a PR with instructions.

Evan

[1] https://github.com/ekfriis/FinalStateAnalysis/blob/master/Utilities/python/cfgtools.py
[2] http://cmssw.cvs.cern.ch/cgi-bin/cmssw.cgi/CMSSW/EgammaAnalysis/ElectronTools/python/calibratedPatElectrons_cfi.py?revision=1.16&view=markup

File: PatTools/python/patTupleProduction.py [hunk 2]
```diff
@@ -330,11 +330,11 @@ def configurePatTuple(process, isMC=True, **kwargs):
     process.customizeElectronSequence.insert(0, process.selectedPatJets)
     process.cleanPatElectrons.src = final_electron_collection
     #setup the energy regression for the specific dataset
-    process.patElectronEnergyCorrections.isMC = cms.bool(bool(isMC))
-    process.patElectronEnergyCorrections.isAOD = \
-        cms.bool(bool(kwargs['isAOD']))
-    process.patElectronEnergyCorrections.dataSet = \
-        cms.string(kwargs['calibrationTarget'])
```
Comment (ekfriis) [inline PatTools/python/patTupleProduction.py:337]: You will still need to modify the calibration target depending on the sample, correct?

Comment (dabelknap) [inline PatTools/python/patTupleProduction.py:337]: I don't think we are using `calibrationTarget` specifically. But, there is a variable called `process.calibratedPatElectrons.inputDataset` which is either "Summer12_LegacyPaper" or "22Jan2013ReReco". I need to add this.

**Comment by ekfriis** [comment]
Hi Austin, 

I merged your progress into my PR:

https://github.com/uwcms/FinalStateAnalysis/pull/209

and also added the functionality needed for you to fix the input tag problem, see the linked PR.

Evan

**Pull request merged**